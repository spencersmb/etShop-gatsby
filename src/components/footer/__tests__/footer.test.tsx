import Footer from '@components/footer/footer'
import React from 'react'
import renderer from 'react-test-renderer'
import {
	render,
	cleanup
} from 'react-testing-library'
import 'jest-dom/extend-expect'

afterEach(cleanup)

describe('Footer Tests', () => {

	it('Should render Logo', () => {
		const modalRender = render(<Footer productPage={false}/>)
		const element = modalRender.getByTestId('footer-logo')
		expect(element).toBeTruthy()
	})

	it('Should render correct nav links', () => {
		const modalRender = render(<Footer productPage={false}/>)
		const element = modalRender.getByTestId('footer-links')
		expect(element).toBeTruthy()
		expect(element.children.length).toEqual(4)
		const items = element.children
		const link1 = items[0].children[0]
		const link2 = items[1].children[0]
		const link3 = items[2].children[0]
		const link4 = items[3].children[0]

		expect(link1.innerHTML).toBe('Products')
		expect(link1.getAttribute('href')).toBe('/products')

		expect(link2.innerHTML).toBe('Blog')
		expect(link2.getAttribute('href')).toBe('https://every-tuesday.com')

		expect(link3.innerHTML).toBe('Support')
		expect(link3.getAttribute('href')).toBe('/support')

		expect(link4.innerHTML).toBe('Privacy Policy')
		expect(link4.getAttribute('href')).toBe('/privacy')
	})

	it('Should render correct social links', () => {
		const modalRender = render(<Footer productPage={false}/>)
		const element = modalRender.getByTestId('social-links')
		expect(element).toBeTruthy()
		expect(element.children.length).toEqual(5)
		const items = element.children
		const link1 = items[0].children[0]
		const link2 = items[1].children[0]
		const link3 = items[2].children[0]
		const link4 = items[3].children[0]

		expect(link1.getAttribute('href')).toBe('https://www.youtube.com/user/everytues')
		expect(link2.getAttribute('href')).toBe('https://www.instagram.com/everytuesday/')
		expect(link3.getAttribute('href')).toBe('https://www.pinterest.com/teelac/')
		expect(link4.getAttribute('href')).toBe('https://www.facebook.com/everytues')
	})

})
