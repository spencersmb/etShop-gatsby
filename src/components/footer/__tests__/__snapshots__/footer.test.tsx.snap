// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Tests renders correctly 1`] = `
<footer
  className="sc-iwsKbI iIYczS"
>
  <div
    className="sc-gZMcBi hElqkJ"
  >
    <h4>
      Get notified
    </h4>
    <p>
      when I release new products, send out promos or freebies.
    </p>
  </div>
  <form
    className="sc-EHOje gxOBQB"
    data-testid="emailBox"
    onSubmit={[Function]}
  >
    <div
      className="sc-bZQynM hQazJB"
    >
      <div
        className="sc-gzVnrw kmLGqm"
      >
        <test-file-stub />
      </div>
      <input
        data-testid="email-input"
        onChange={[Function]}
        placeholder="Enter your email address"
        required={true}
        type="email"
      />
    </div>
    <div
      className="sc-htoDjs bhZFqV"
    >
      <div
        className="sc-ifAKCX kIktAd"
        data-testid="submitBtn"
      >
        <button
          data-testid="button"
          disabled={false}
          onClick={[Function]}
        >
          <div
            className="buttonText"
          >
            Submit
          </div>
        </button>
      </div>
    </div>
    <div
      className="sc-dnqmqq jgwIwO"
    >
      <test-file-stub />
    </div>
  </form>
  <nav
    className="sc-VigVT fEPntw"
  >
    <div
      className="sc-gqjmRU evmKLe"
    >
      <test-file-stub />
    </div>
    <div
      className="sc-jTzLTM kIerjd"
    >
      <div
        className="sc-fjdhpX cyLcxw"
        data-testid="footer-logo"
      >
        <test-file-stub />
      </div>
      <ul
        className="sc-jzJRlG fOqSXj"
        data-testid="footer-links"
      >
        <li>
          <a
            href="/products"
          >
            Products
          </a>
        </li>
        <li>
          <a
            href="https://every-tuesday.com"
            rel="noreferrer"
            target="_blank"
          >
            Blog
          </a>
        </li>
        <li>
          <a
            href="/support"
          >
            Support
          </a>
        </li>
        <li>
          <a
            href="/privacy"
          >
            Privacy Policy
          </a>
        </li>
      </ul>
      <ul
        className="sc-cSHVUG bctMro"
        data-testid="social-links"
      >
        <li>
          <a
            href="https://www.youtube.com/user/everytues"
            rel="noreferrer"
            target="_blank"
          >
            <test-file-stub />
          </a>
        </li>
        <li>
          <a
            href="https://www.instagram.com/everytuesday/"
            rel="noreferrer"
            target="_blank"
          >
            <test-file-stub />
          </a>
        </li>
        <li>
          <a
            href="https://www.pinterest.com/teelac/"
            rel="noreferrer"
            target="_blank"
          >
            <test-file-stub />
          </a>
        </li>
        <li>
          <a
            href="https://www.facebook.com/everytues"
            rel="noreferrer"
            target="_blank"
          >
            <test-file-stub />
          </a>
        </li>
        <li>
          <a
            href="https://twitter.com/teelacunningham"
            rel="noreferrer"
            target="_blank"
          >
            <test-file-stub />
          </a>
        </li>
      </ul>
    </div>
  </nav>
</footer>
`;
